dhcpv4/utils.go

```diff
@@ -0,0 +1,14 @@
+package dhcpv4
+
+// IsRequested function takes a DHCPv4 message and an OptionCode, and returns
+// true if that option is within the requested options of the DHCPv4 message.
+func IsRequested(pkt *DHCPv4, requested OptionCode) bool {
+	for _, optprl := range pkt.GetOption(OptionParameterRequestList) {
```

Comment:
in general we don't validate if more than one option of the same type is found in a packet. Also most DHCP implementations stop looking after the first option of a given type. 
Shall we make this more opinionated and only look in the first option only? I.e. using `GetOneOption` instead of `GetOption`

Reply:
For v6 I have seen clients sending more than 1 requested option, each one requesting different things. I'll rather check all of them.

Reply:
oh wow. OK. And the RFC doesn't make it easy either:

>   Unless otherwise noted, each option may appear only in the options
>   area of a DHCP message and may appear only once.  If an option does
>   appear multiple times, each instance is considered separate and the
>   data areas of the options MUST NOT be concatenated or otherwise
>   combined.